js/copy-service/Parser/Parser.ts

```diff
@@ -1,100 +1,97 @@
-  ARGS_START: '}[',
-  ARGS_COMMA: ',',
-  ARGS_END: ']',
-  NEWLINE: '\n',
-  WORD_BREAK: '\b'
-};
+class Parser {
+  /**
+   * The supported tokens in copy.
+   *
+   * @type {object}
+   */
+  static TOKENS = {
+    TEXT: 'text',
+    SWITCH_DELIM: '}{',
+    CLOSE: '}',
+    REF_START: '${',
+    SUB_START: '\#{', // eslint-disable-line no-useless-escape
+    REF_SUB_START: '%{',
+    SWITCH_START: '*{',
+    FUNC_START: '^{',
+    HTML_TAG_START: '<',
+    HTML_TAG_END: '>',
+    ARGS_START: '}[',
+    ARGS_COMMA: ',',
+    ARGS_END: ']',
+    NEWLINE: '\n',
+    WORD_BREAK: '\b'
+  } as const;
 
-/**
- * All TOKENS that are not TEXT, TAG, or ARGS tokens.
- *
- * @type {Array}
- */
-const NON_TEXT_TOKENS = _.filter(_.values(TOKENS), (token) => (
-  !_.includes([
-    TOKENS.TEXT,
-    TOKENS.HTML_TAG_START,
-    TOKENS.HTML_TAG_END,
-    TOKENS.ARGS_START,
-    TOKENS.ARGS_COMMA,
-    TOKENS.ARGS_END
-  ], token)
-));
+  /**
+   * All TOKENS that are not TEXT, TAG, or ARGS tokens.
+   *
+   * @type {Array}
+   */
+  static NON_TEXT_TOKENS = _.filter(_.values(this.TOKENS), (token) => (
+    !_.includes([
+      this.TOKENS.TEXT,
+      this.TOKENS.HTML_TAG_START,
+      this.TOKENS.HTML_TAG_END,
+      this.TOKENS.ARGS_START,
+      this.TOKENS.ARGS_COMMA,
+      this.TOKENS.ARGS_END
+    ], token)
+  ));
 
-/**
- * RegExp for the starting tag of allowed HTML tags.
- *
- * @type {RegExp}
- */
-const HTML_START_TAG_REGEX = /^<(\w+)>/;
+  /**
+   * RegExp for the starting tag of allowed HTML tags.
+   *
+   * @type {RegExp}
+   */
+  static HTML_START_TAG_REGEX = /^<(\w+)>/;
 
-/**
- * RegExp for the ending tag of allowed HTML tags.
- *
- * @type {RegExp}
- */
-const HTML_END_TAG_REGEX = /^<\/(\w+)>/;
+  /**
+   * RegExp for the ending tag of allowed HTML tags.
+   *
+   * @type {RegExp}
+   */
+  static HTML_END_TAG_REGEX = /^<\/(\w+)>/;
 
-/**
- * The supported HTML tags in copy.
- *
- * @type {Array}
- */
-const ALLOWED_HTML_TAGS = [
-  // cspell:disable-next-line
-  'u', 'sup', 'sub', 's', 'em', 'strong', 'p', 'span', 'div', 'ol', 'ul', 'li', 'b', 'i', 'u'
-];
+  /**
+   * The supported HTML tags in copy.
+   *
+   * @type {Array}
+   */
+  static ALLOWED_HTML_TAGS = [
+    // cspell:disable-next-line
+    'u', 'sup', 'sub', 's', 'em', 'strong', 'p', 'span', 'div', 'ol', 'ul', 'li', 'b', 'i', 'u'
+  ] as const;
 
-/**
- * The supported delimiter for subkeys.
- *
- * @type {string}
- */
-const KEY_DELIMITER = '.';
+  /**
+   * The supported delimiter for subkeys.
+   *
+   * @type {string}
+   */
+  static KEY_DELIMITER = '.';
 
-/**
- * Parses raw json copy into ASTs.
- */
-class Parser {
   /**
    * Transforms raw copy into ASTs. Will mutate the `tree` argument.
-   *
-   * @param  {object} tree
-   * @returns {object} The same object, with leaves parsed to ASTs.
    */
-  static parseLeaves(tree) {
-    _.forEach(tree, (node, key) => {
+  static parseLeaves(tree: { [key: string]: any; }): { [key: string]: SyntaxNode; } | never {
```

Comment:
I think this is what you want

```suggestion
  type leafParam = SyntaxNode | string | { [key: string]: leafParam };

  static parseLeaves(tree: leafParam): { [key: string]: SyntaxNode; } | never {
```